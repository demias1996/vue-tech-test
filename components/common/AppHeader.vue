<template>
    <header>
        <b-navbar toggleable="lg" type="dark" variant="dark">
            <b-icon-bookmarks-fill class="mr-2" font-scale="1.75" variant="light" />
            <b-navbar-brand class="header-title" href="#">{{ title }}</b-navbar-brand>

            <b-navbar-toggle target="nav-collapse" />

            <b-collapse id="nav-collapse" is-nav>
                <b-navbar-nav>
                    <li class="nav-item">
                        <nuxt-link class="nav-link" to="/">Home</nuxt-link>
                    </li>
                </b-navbar-nav>

                <b-navbar-nav class="ml-auto">
                    <b-nav-form class='search-form' @submit.prevent="performSearch">
                        <b-form-input
                            v-model="searchQuery"
                            size="sm"
                            class="mr-sm-2"
                            placeholder="Search"
                        />
                        <b-button
                            size="sm"
                            class="search-button my-2 my-sm-0"
                            type="submit"
                        >
                            Search
                        </b-button>
                    </b-nav-form>
                </b-navbar-nav>
            </b-collapse>
        </b-navbar>
    </header>
</template>

<script>
    import { BIconBookmarksFill } from 'bootstrap-vue';

    export default {
        name: 'app-header',

        components: {
            BIconBookmarksFill,
        },

        props: {
            title: {
                type: String,
                default: '',
            },
        },

        data() {
            return {
                searchQuery: '',
            };
        },

        methods: {
            /**
             * Redirects to the main page with search query param.
             */
            performSearch() {
                this.$router.push(`/?searchQuery=${ this.searchQuery }`);
            },
        },
    };
</script>
