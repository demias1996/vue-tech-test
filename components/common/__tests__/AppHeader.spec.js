import { shallowMount } from '@vue/test-utils';
import AppHeader from '@/components/common/AppHeader';

describe('AppHeader', () => {
    test('is a Vue instance', () => {
        const wrapper = shallowMount(AppHeader);
        expect(wrapper.vm).toBeTruthy();
    });

    test('is rendered correctly with default props', () => {
        const wrapper = shallowMount(AppHeader);
        expect(wrapper.element).toMatchSnapshot();
    });

    test('title to be displayed', () => {
        const wrapper = shallowMount(AppHeader, {
            propsData: {
                title: 'Test title',
            },
        });

        const titleElement = wrapper.find('.header-title');

        expect(titleElement.text()).toBe('Test title');
        expect(wrapper.element).toMatchSnapshot();
    });

    test('search is performed', () => {
        const pushStub = jest.fn();
        const wrapper = shallowMount(AppHeader, {
            mocks: {
                $router: {
                    push: pushStub,
                },
            },
        });

        wrapper.vm.searchQuery = 'Test search';

        const searchForm = wrapper.find('.search-form');

        searchForm.trigger('submit');

        expect(pushStub).toHaveBeenCalledTimes(1);
        expect(pushStub).toHaveBeenCalledWith('/?searchQuery=Test search');
    });
});
