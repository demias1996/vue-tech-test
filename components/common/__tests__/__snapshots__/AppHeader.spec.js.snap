// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader is rendered correctly with default props 1`] = `
<header>
  <b-navbar-stub
    tag="nav"
    toggleable="lg"
    type="dark"
    variant="dark"
  >
    <b-icon-bookmarks-fill-stub
      class="mr-2"
      fontscale="1.75"
      rotate="0"
      scale="1"
      shifth="0"
      shiftv="0"
      variant="light"
    />
     
    <b-navbar-brand-stub
      class="header-title"
      href="#"
      tag="div"
      target="_self"
    >
      
    </b-navbar-brand-stub>
     
    <b-navbar-toggle-stub
      label="Toggle navigation"
      target="nav-collapse"
    />
     
    <b-collapse-stub
      id="nav-collapse"
      isnav="true"
      tag="div"
    >
      <b-navbar-nav-stub
        tag="ul"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            to="/"
          >
            Home
          </a>
        </li>
      </b-navbar-nav-stub>
       
      <b-navbar-nav-stub
        class="ml-auto"
        tag="ul"
      >
        <b-nav-form-stub
          class="search-form"
        >
          <b-form-input-stub
            class="mr-sm-2"
            debounce="0"
            formatter="[Function]"
            placeholder="Search"
            size="sm"
            type="text"
            value=""
          />
           
          <b-button-stub
            class="search-button my-2 my-sm-0"
            size="sm"
            tag="button"
            target="_self"
            type="submit"
            variant="secondary"
          >
            
                        Search
                    
          </b-button-stub>
        </b-nav-form-stub>
      </b-navbar-nav-stub>
    </b-collapse-stub>
  </b-navbar-stub>
</header>
`;

exports[`AppHeader title to be displayed 1`] = `
<header>
  <b-navbar-stub
    tag="nav"
    toggleable="lg"
    type="dark"
    variant="dark"
  >
    <b-icon-bookmarks-fill-stub
      class="mr-2"
      fontscale="1.75"
      rotate="0"
      scale="1"
      shifth="0"
      shiftv="0"
      variant="light"
    />
     
    <b-navbar-brand-stub
      class="header-title"
      href="#"
      tag="div"
      target="_self"
    >
      Test title
    </b-navbar-brand-stub>
     
    <b-navbar-toggle-stub
      label="Toggle navigation"
      target="nav-collapse"
    />
     
    <b-collapse-stub
      id="nav-collapse"
      isnav="true"
      tag="div"
    >
      <b-navbar-nav-stub
        tag="ul"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            to="/"
          >
            Home
          </a>
        </li>
      </b-navbar-nav-stub>
       
      <b-navbar-nav-stub
        class="ml-auto"
        tag="ul"
      >
        <b-nav-form-stub
          class="search-form"
        >
          <b-form-input-stub
            class="mr-sm-2"
            debounce="0"
            formatter="[Function]"
            placeholder="Search"
            size="sm"
            type="text"
            value=""
          />
           
          <b-button-stub
            class="search-button my-2 my-sm-0"
            size="sm"
            tag="button"
            target="_self"
            type="submit"
            variant="secondary"
          >
            
                        Search
                    
          </b-button-stub>
        </b-nav-form-stub>
      </b-navbar-nav-stub>
    </b-collapse-stub>
  </b-navbar-stub>
</header>
`;
